app/src/main-process/main.ts

```diff
@@ -104,6 +105,10 @@ if (__WIN32__ && process.argv.length > 1) {
   }
 }
 
+if (shellNeedsPatching(process)) {
+  getEnvironmentFromShell(mergeEnvironmentVariables)
```

Comment:
Given that we're using the same callback function, it seems like we could hide that in `lib/shell`?